### PR TITLE
[DOCS] Add dataset progress docs

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

**Type of change**

- Documentation update

**How Has This Been Tested**

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: argilla/docs/how_to_guides/distribution.md [hunk 1]
```diff
@@ -77,4 +77,55 @@ dataset = client.datasets("my_dataset")
 dataset.settings.distribution.min_submitted = 4
 
 dataset.update()
-```
\ No newline at end of file
+```
+
+## Dataset progress
```
Comment (nataliaElv) [inline argilla/docs/how_to_guides/distribution.md]:
```suggestion
## Track your team's progress
```

Comment (nataliaElv) [reply argilla/docs/how_to_guides/distribution.md]: Note that if this change is accepted, you'll need to correct the link to this section in the "Annotate the dataset" page.

**Review by nataliaElv — COMMENTED** [review]
In the API reference (https://docs.argilla.io/docs_add-dataset-progress-docs/reference/argilla/datasets/datasets/#src.argilla.datasets._resource.Dataset.progress)
- The text "Returns the progress of the dataset creation." doesn't seem to be an exact description of what the method does. I would change it for something like "Returns the progress made on the dataset." or "Returns the team's progress on the dataset.". 
- The "Returns" table seems a bit off (screenshot attached). I guess it should be one row only.
Captura de pantalla 2024-09-12 a las 12 41 41

**Review by sdiazlor — APPROVED** [review]
Apart from Natalia's comments, LGTM!

File: argilla/src/argilla/datasets/_resource.py [hunk 2]
```diff
@@ -179,21 +179,45 @@ def progress(self, with_users_distribution: bool = False) -> dict:
 
         Parameters:
             with_users_distribution (bool): If True, the progress of the dataset is returned
-                with users distribution.
+                with users distribution. This includes the number of completed submissions for each user.
```
Comment (nataliaElv) [inline argilla/src/argilla/datasets/_resource.py]:
```suggestion
                with users distribution. This includes the number of responses made by each user.
```

Comment (nataliaElv) [reply argilla/src/argilla/datasets/_resource.py]: Also see the comment that I made earlier on the "Returns the progress..." text on the API reference:
https://github.com/argilla-io/argilla/pull/5481#pullrequestreview-2299882811

File: argilla/src/argilla/datasets/_resource.py [hunk 3]
```diff
@@ -179,21 +179,45 @@ def progress(self, with_users_distribution: bool = False) -> dict:
 
         Parameters:
             with_users_distribution (bool): If True, the progress of the dataset is returned
-                with users distribution.
+                with users distribution. This includes the number of responses made by each user.
 
         Returns:
             dict: The progress of the dataset creation
+
+        An example of response when `with_users_distribution` is `True`:
```
Comment (frascuchon) [inline argilla/src/argilla/datasets/_resource.py]:
```suggestion
        Returns:
            dict: The team's progress on the dataset.

        An example of a response when `with_users_distribution` is `True`:
```

**Review by nataliaElv — APPROVED** [review]

**Pull request merged**